linkerd/app/outbound/src/lib.rs

```diff
@@ -233,12 +237,18 @@ impl<A: OrigDstAddr> Config<A> {
                     .push(http::timeout::layer())
                     .push(metrics.http_route.into_layer::<classify::Response>())
                     .push(classify::Layer::new())
-                    .push_buffer_pending(buffer.max_in_flight, DispatchDeadline::extract);
+                    .push_into_new_service()
+                    .push_on_response(
+                        svc::layers().push_buffer(buffer.max_in_flight, DispatchDeadline::extract),
+                    );
 
                 distributor
-                    .serves::<DstAddr>()
-                    .push_buffer_pending(buffer.max_in_flight, DispatchDeadline::extract)
-                    .makes::<DstAddr>()
+                    .check_service::<DstAddr>()
+                    .into_new_service()
+                    .push_on_response(
+                        svc::layers().push_buffer(buffer.max_in_flight, DispatchDeadline::extract),
+                    )
```

Comment:
Is there a difference between 
```rust
.push_on_response(
    svc::layers().push_buffer(buffer.max_in_flight, DispatchDeadline::extract),
)
```
and 
```rust
.push_on_response(
    buffer::layer(buffer.max_in_flight, DispatchDeadline::extract)
)
```
it seems like these _should_ be equivalent, but I wanted to be sure this is correct?

Reply:
they're the same